transmission_interface/src/bidirectional_effort_joint_interface_provider.cpp

```diff
@@ -0,0 +1,87 @@
+//     documentation and/or other materials provided with the distribution.
+//   * Neither the name of PAL Robotics S.L. nor the names of its
+//     contributors may be used to endorse or promote products derived from
+//     this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+//////////////////////////////////////////////////////////////////////////////
+
+// Pluginlib
+#include <pluginlib/class_list_macros.h>
+
+// ros_control
+#include <transmission_interface/bidirectional_effort_joint_interface_provider.h>
+
+namespace transmission_interface
+{
+    bool BiDirectionalEffortJointInterfaceProvider::registerTransmission(TransmissionLoaderData &loader_data,
+            TransmissionHandleData &handle_data)
+    {
+        // Setup joint state interface first (if not yet done)
+        if(!hasResource(handle_data.name, loader_data.transmission_interfaces.act_to_jnt_state))
+        {
+            if(!JointStateInterfaceProvider::registerTransmission(loader_data, handle_data))
+            {
+                return false;
+            }
+        }
+
+        if(!hasResource(handle_data.name, loader_data.inverse_transmission_interfaces.jnt_to_act_state))
+        {
+            if(!loader_data.robot_transmissions->get<transmission_interface::JointToActuatorStateInterface>())
+            {
+                loader_data.robot_transmissions->registerInterface(&loader_data.inverse_transmission_interfaces.jnt_to_act_state);
+            }
+
+            transmission_interface::JointToActuatorStateInterface &interface = *(loader_data.robot_transmissions->get<transmission_interface::JointToActuatorStateInterface>());
+
+            // Update transmission interface
+            transmission_interface::JointToActuatorStateHandle handle(handle_data.name, handle_data.transmission.get(), handle_data.act_state_data, handle_data.jnt_state_data);
+            interface.registerHandle(handle);
+        }
+
+        // If command interface does not yet exist in the robot transmissions, add it and use internal data structures
+        if(!loader_data.robot_transmissions->get<transmission_interface::JointToActuatorEffortInterface>())
```

Comment:
This `if` block is here: https://github.com/jlack1987/ros_control/blob/dfcb72375a3378b99e64971a46ef5f612f7dbbca/transmission_interface/src/effort_joint_interface_provider.cpp#L130